src/join/composer.jl

```diff
@@ -55,8 +56,25 @@ struct DataFrameJoiner
                                         "when matchmissing == :error"))
                 end
             end
+        elseif matchmissing === :notequal
+            if kind in (:left, :semi, :anti)
+                dfr = dropmissing(dfr, right_on, view=true)
+                dfr_on = select(dfr, right_on)
```

Comment:
Wouldn't `copycols=false` make sense here?

Reply:
definitely - same below. Thank you for spotting this.

Similarly lines 49 and 50 above duplicate work so it is better to do this operation only once (this is a minor issue), https://github.com/JuliaData/DataFrames.jl/pull/2724/files#diff-00fe5953fad7e5572a8a4ae57ab3724fd3a3895cae501c40bc60a9fed12cf60fR49-R50.

Reply:
Regarding lines 49-50
I noticed it but I had trouble with coming up with some elegant solution without obscuring ifs.
Since always view is returned here I decided to leave it as is.
After you comments I'll change it for `select(dfl, left_on, copycols=true)` for consistency.

Reply:
> I noticed it but I had trouble with coming up with some elegant solution without obscuring ifs.

The way would be:
```
if matchmissing === :notequal
    # do the stuff you have added and in every place explicitly define `dfl_on` and dfr_on`
else
    if matchmissing === :error
        for df in (dfl_on, dfr_on), col in eachcol(df)
                if any(ismissing, col)
                    throw(ArgumentError("missing values in key columns are not allowed " *
                                        "when matchmissing == :error"))
                end
            end
    end
    dfl_on = dfl[!, left_on]
    dfr_on = dfr[!, right_on]
end
```
in this way you define `dfl_on` and `dfr_on` only once.

Reply:
But since it is used later in line 82, I will have to add `local dfl_on` before `if`, right?

Reply:
no, because `if` does not introduce a new scope, see https://docs.julialang.org/en/v1/manual/variables-and-scoping/#man-scope-table

Reply:
As an example why not having `if` create a new scope is useful consider e.g. this part of source: https://github.com/JuliaData/DataFrames.jl/blob/f56982d063254e947ca98247e5f471f8a095ecb0/src/DataFrames.jl#L92.

As you can see we either import or define `@spawn` conditionally, but it all happens in global scope. If `if` created a local scope then such things would be much harder to write.

Reply:
Very interesting! I assumed it's like in c++ for some reason, I need to expand my knowledge on this topic.
Thanks